edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -15,10 +15,24 @@ public class DockerConfig : IEquatable<DockerConfig>
     {
         // This is not the actual docker image regex, but a less strict version.
         const string ImageRegexPattern = @"^(?<repo>([^/]*/)*)(?<image>[^/:]+)(?<tag>:[^/:]+)?$";
+        const string ImageUpstreamRegexPattern = @"^\$(?<host>.*?)(?<path>:.*)";
```

Comment:
I need to understand more here.  It seems what host you are going to match is same as repo.  is it possible to combine regex into 1?

Reply:
I think you need to handle the case if image address don't have any port mentioned, e.g. $IOTEDGEGATEWAY_HOSTNAME/comp/ea:tag1.

Reply:
Yes, I think I can combine them. I'll look into it. I was not so much in favor because it is a different logic.
By design a port has to be specified. If a port is missing we dont do anything.

Reply:
After giving it some thought I think it is less confusing to separate them. The regex is hard to read has it is already. It is less error prone to keep the regex simple and focused on one task.